src/internal/analytics/components/analytics-funnel.tsx

```diff
@@ -247,5 +257,49 @@ export const AnalyticsFunnelSubStep = ({ children }: AnalyticsFunnelSubStepProps
 
   const context = isNested ? inheritedContext : newContext;
 
+  useEffect(() => {
+    const onMouseDown = () => (mousePressed.current = true);
+
+    const onMouseUp = async () => {
+      mousePressed.current = false;
+
+      if (!isFocusedSubStep.current) {
+        return;
+      }
+
+      /*
+        Some mouse events result in an element being focused. However,
+        this happens only _after_ the onMouseUp event. We yield the
+        event loop here, so that `document.activeElement` has the
+        correct new value.      
+      */
+      await new Promise(r => setTimeout(r, 1));
+
+      if (!subStepRef.current || !subStepRef.current.contains(document.activeElement)) {
+        isFocusedSubStep.current = false;
```

Comment:
Is this necessary to set this one to false here? I would imagine in the next similar flow it is going to be set to true anyways. If not, wouldn't it make sense to reset this outside of the condition block and also in the effect cleanup function?

Reply:
`isFocusedSubStep.current` is checked in a separate file when an element in the substep is focused, so that the `onFocus` handling runs when the focus _enters_ the substep, but not when it moves around _inside_ the substep:
https://github.com/cloudscape-design/components/blob/3482439f8affd1a0d69be1009267db229b800dd0/src/internal/analytics/hooks/use-funnel.ts#L44-L54

If we do not set the variable to `false` here, then the next time the user enters this substep we will miss emitting the `funnelSubStepStart` event.

Reply:
Ah, so when there was a click inside the isFocusedSubStep shall remain true so that the focus is ignored?

Reply:
Yes. The `onFocus` event listener is attached to the whole substep, and it fires every time that any element inside the container receives the focus, because the `onFocus` event bubbles (only in React). So we want to ignore all subsequent `onFocus` events if the focus is _already_ somewhere inside the substep.